sunpy/io/file_tools.py

```diff
@@ -214,14 +214,12 @@ def detect_filetype(filepath):
         cdf_magic_number = fp.read(4).hex()
 
     # FITS
-    # Check the extensions to see if it is a gzipped FITS file
-    filepath_rest_ext1, ext1 = os.path.splitext(filepath)
-    _, ext2 = os.path.splitext(filepath_rest_ext1)
-
-    gzip_extensions = [".gz"]
-    fits_extensions = [".fts", ".fit", ".fits"]
-    if (ext1 in gzip_extensions and ext2 in fits_extensions):
-        return 'fits'
+    # Checks for gzip signature. 
+    # If found, decompresses first few bytes and checks for FITS
+    if first80[:3] == b"\x1f\x8b\x08":
+        import gzip
```

Comment:
I don't think there's a lot to gain by putting this here, we can move it to the top of the file.